arches/app/media/js/viewmodels/workflow-step.js

```diff
@@ -37,6 +37,15 @@ define([
         this.postSaveCallback = ko.observable();
 
         this.externalStepData = {};
+        this.locked = ko.observable();
+
+        this.setLock = function(stepName, locked) {
```

Comment:
small nit: can this function be moved to below the initialize method, with the other function declarations?